typelib/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@ endif (NOT TYPELIB_HARDCODED_PLUGIN_PATH)
 set_source_files_properties(endianness.cc PROPERTIES
     COMPILE_FLAGS -fno-strict-aliasing)
 
-target_link_libraries(typeLib ${CMAKE_DL_LIBS} ${Boost_FILESYSTEM_LIBRARIES} ${Boost_SYSTEM_LIBRARIES})
+target_link_libraries(typeLib pthread ${CMAKE_DL_LIBS} ${Boost_FILESYSTEM_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${Boost_THREAD_LIBRARIES})
```

Comment:
Boost mutexes are used in [typelib/utilmm/singleton/server.hh](https://github.com/orocos-toolchain/typelib/blob/69476e5e886138d1626dc234a38198d89d5063cb/typelib/utilmm/singleton/server.hh#L9) and hence the library must be linked to [Boost Thread](https://www.boost.org/doc/libs/1_75_0/doc/html/thread.html). Could be that `boost::recursive_mutex` is header-only if compiled in Release mode, but we observed linker errors in the past, at least in `Debug` mode.